### PR TITLE
version: more explicit version for comparison

### PR DESCRIPTION
**Pull request by vbatts**
For deterministic comparison and still providing a Version string for
representation and reuse.

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -1,4 +1,15 @@
 package specs
 
+import "fmt"
+
+const (
+	// VersionMajor is for an API incompatible changes
+	VersionMajor = 0
+	// VersionMinor is for functionality in a backwards-compatible manner
```
Comment (wking) [inline version.go]: Maybe “VersionMinor is for backwards-compatible API changes”?  And then mirror that for `VersionMajor` with “VersionMajor is for backwards-incompatible API changes”.

File: version_test.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package specs
+
+import (
+	"regexp"
+	"testing"
+)
+
+func TestVersion(t *testing.T) {
+	re := regexp.MustCompile(`\d+\.\d+\.\d+`)
```
Comment (wking) [inline version_test.go]: SemVer 2.0 also allows for pre-release versions and build metadata.  See [sections 9 and 10][1].  Surely someone else has already written a SemVer validator in Go?

[1]: http://semver.org/spec/v2.0.0.html

File: version.go (hunk 1 above)
Comment (vbatts) [inline version.go]: this is leaning on verbiage from SemVer

File: version_test.go (hunk 2 above)
Comment (vbatts) [inline version_test.go]: This _test.go can be dropped entirely. It's just a smoke test and not intended to be a SemVer complete assertion.

File: version.go (hunk 1 above)
Comment (wking) [inline version.go]: On Thu, Sep 10, 2015 at 12:32:25PM -0700, Vincent Batts wrote:
> > +	// VersionMinor is for functionality in a backwards-compatible manner
> 
> this is leaning on verbiage from SemVer

Ah, consistency with semver.org is probably better than consistency
between the entries in this case.  So suggestion retracted ;).

File: version_test.go (hunk 2 above)
Comment (wking) [inline version_test.go]: On Thu, Sep 10, 2015 at 12:33:20PM -0700, Vincent Batts wrote:
> > +	re := regexp.MustCompile(`\d+\.\d+\.\d+`)
> 
> This _test.go can be dropped entirely…

That's what I'd recommend.

Comment (philips) [inline version_test.go]: @vbatts in the future we can do this: https://github.com/coreos/go-semver

**Comment by philips** [comment]
I don't know if this helps anything at this point. We should just make a note in the README that says:

"During the 0.* series of OCI releases we make no backwards compatibility guarantees and intend to break the schema during this series."

**Comment by vbatts** [comment]
On Sep 10, 2015 3:48 PM, "Brandon Philips" <notifications@github.com> wrote:
>
> I don't know if this helps anything at this point. We should just make a
note in the README that says:
>
> "During the 0.* series of OCI releases we make no backwards compatibility
guarantees and intend to break the schema during this series."
>

Totally agree and was just fielding that response on irc

File: version_test.go (hunk 2 above)
Comment (vbatts) [inline version_test.go]: @philips right on.

**Comment by vbatts** [comment]
updated

removed the test and added release verbiage

**Comment by wking** [comment]
On Thu, Sep 10, 2015 at 01:18:47PM -0700, Vincent Batts wrote:
> updated

Through d335ecd looks good to me.

**Comment by crosbymichael** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**